src/test/java/net/sf/jabref/importer/CustomImporterTest.java

```diff
@@ -0,0 +1,106 @@
+import java.util.List;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.importer.fileformat.BibTeXMLImporter;
+import net.sf.jabref.importer.fileformat.CopacImporter;
+import net.sf.jabref.importer.fileformat.OvidImporter;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class CustomImporterTest {
+
+    private CustomImporter importer;
+
+    @Before
+    public void setUp() {
+        Globals.prefs = JabRefPreferences.getInstance();
+
+        BibTeXMLImporter bibtexml = new BibTeXMLImporter();
+        importer = new CustomImporter();
+        importer.setName(bibtexml.getFormatName());
+        importer.setCliId(bibtexml.getId());
+        importer.setClassName(bibtexml.getClass().getName());
+        importer.setBasePath("src/main/java/net/sf/jabref/importer/fileformat/BibTeXMLImporter.java");
+    }
+
+    @Test
+    public void testGetName() {
+        assertEquals("BibTeXML", importer.getName());
+    }
+
+    @Test
+    public void testGetCliId() {
+        assertEquals("bibtexml", importer.getClidId());
+    }
+
+    @Test
+    public void testGetClassName() {
+        assertEquals("net.sf.jabref.importer.fileformat.BibTeXMLImporter", importer.getClassName());
+    }
+
+    @Test
+    public void testGetBasePath() {
+        assertEquals("src/main/java/net/sf/jabref/importer/fileformat/BibTeXMLImporter.java", importer.getBasePath());
+    }
+
+    @Test
+    public void testGetInstance() throws IOException, ClassNotFoundException,
+            InstantiationException, IllegalAccessException {
+        assertEquals(new BibTeXMLImporter(), importer.getInstance());
+    }
+
+    @Test
+    public void testGetFileFromBasePath() {
+        assertEquals(new File(importer.getBasePath()), importer.getFileFromBasePath());
+    }
+
+    @Test
+    public void testGetBasePathUrl() throws MalformedURLException {
+        assertEquals(new File(importer.getBasePath()).toURI().toURL(), importer.getBasePathUrl());
+    }
+
+    @Test
+    public void testGetAsStringList() {
+        assertEquals(importer.getName(), importer.getAsStringList().get(0));
+        assertEquals(importer.getClidId(), importer.getAsStringList().get(1));
+        assertEquals(importer.getClassName(), importer.getAsStringList().get(2));
+        assertEquals(importer.getBasePath(), importer.getAsStringList().get(3));
+    }
+
+    @Test
+    public void testEquals() {
+        boolean test1 = importer.equals(importer);
+        boolean test2 = importer.equals(new CopacImporter());
+
+        assertTrue(test1);
+        assertFalse(test2);
+    }
+
+    @Test
+    public void testCompareTo() {
+        OvidImporter ovid = new OvidImporter();
+        List<String> dataTest = Arrays.asList(ovid.getFormatName(), ovid.getId(), ovid.getClass().getName(), "src/main/java/net/sf/jabref/importer/fileformat/OvidImporter.java");
+
+        CopacImporter copac = new CopacImporter();
```

Comment:
Add constructor in CustomImport which wraps around an importer and use it everywhere.